CRM/Contribute/Form/Contribution.php

```diff
@@ -462,6 +462,25 @@ public function setDefaultValues() {
       $defaults['campaign_id'] = $this->_pledgeValues['campaign_id'];
     }
 
+    if ($this->_id) {
+      $financialTrxn = $this->getlatestPayments();
+      if ($financialTrxn) {
+        $paymentProcessorID = CRM_Utils_Array::value('financial_trxn_id.payment_processor_id', $financialTrxn);
+        $result = civicrm_api3('FinancialTrxn', 'getsingle', array(
+          'return' => array("card_type", "pan_truncation"),
+          'id' => $financialTrxn['financial_trxn_id'],
+        ));
+        $defaults['card_type'] = CRM_Utils_Array::value('card_type', $result);
+        $defaults['pan_truncation'] = CRM_Utils_Array::value('pan_truncation', $result);
+
+        if ($paymentProcessorID) {
+          if ($defaults['pan_truncation']) {
+            $defaults['pan_truncation'] = "**** **** **** " . $defaults['pan_truncation'];
+          }
+          $this->freeze(array('card_type', 'pan_truncation'));
```

Comment:
Yeah, we freeze the field if Payment is made from online. The pan truncation is calculated i.e last 4 digit, using credit card number entered by user in payment form of payment processor. But in case of not online payment i.e payment not done via payment processor we allow user to add/update the card type or pan truncation on offline contribution form for credit card payment method.